tera-templates/crate/details.html

```diff
@@ -0,0 +1,303 @@
+                                                    {{ release.version }}
+                                                </a>
+
+                                            {# If the release wasn't yanked but failed to build, show a warning #}
+                                            {%- else -%}
+                                                <a href="/crate/{{ details.name }}/{{ release.version }}" class="pure-menu-link warn"
+                                                    title="docs.rs failed to build {{ details.name }}-{{ release.version }}">
+                                                    <i class="fa fa-fw fa-warning"></i> {{ release.version }}
+                                                </a>
+                                            {%- endif -%}
+                                        </li>
+                                    {%- endfor -%}
+                                </ul>
+                            </div>
+                        </li>
+
+                        {# Display the crate owner's profile picture and a link to their docs.rs profile #}
+                        <li class="pure-menu-heading">Owners</li>
+                        <li class="pure-menu-item">
+                            {%- for owner in details.owners -%}
+                                <a href="/releases/@{{ owner[0] }}">
+                                    <img src="{{ owner[1] }}" alt="{{ owner[0] }}" class="owner">
+                                </a>
+                            {%- endfor -%}
+                        </li>
+                    </ul>
+                </div>
+            </div>
+
+            <div class="pure-u-1 pure-u-sm-17-24 pure-u-md-19-24 package-details" id="main">
+                {# If the release is not a library #}
+                {%- if not details.is_library -%}
+                    <div class="warning">
+                        {{ details.name }}-{{ details.version }} is not a library.
+                    </div>
+
+                {# If the release has been yanked and is a library #}
+                {%- elif details.yanked -%}
+                    <div class="warning">
+                        {{ details.name }}-{{ details.version }} has been yanked.
+                    </div>
+
+                {# If the build succeeded, isn't yanked and is a library #}
+                {%- elif details.build_status -%}
+                    {# If there are no docs display a warning #}
+                    {%- if not details.rustdoc_status -%}
+                        <div class="warning">{{ details.name }}-{{ details.version }} doesn't have any documentation.</div>
+                    {%- endif -%}
+
+                    {# If there's a readme, display it #}
+                    {%- if details.readme -%}
+                        {{ details.readme | safe }}
+
+                    {# If there's not a readme then attempt to display docs #}
```

Comment:
We should probably rename the `rustdoc` variable, it's confusing. I noticed something was off because the docs are HTML, and so shouldn't have `| safe` applied.
```suggestion
                    {# If there's not a readme then display the long_description #}
```

Reply:
`| safe` is required since Tera escapes HTML by default. You get a big ugly mess without the escaping

Reply:
Oh I see. Then this shouldn't have safe applied because long_description is not HTML.